src/ctl.c

```diff
@@ -3457,3 +3467,32 @@ experimental_arenas_i_pactivep_ctl(tsd_t *tsd, const size_t *mib,
 	malloc_mutex_unlock(tsd_tsdn(tsd), &ctl_mtx);
 	return ret;
 }
+
+static int
+experimental_prof_recent_alloc_max_ctl(tsd_t *tsd, const size_t *mib,
+    size_t miblen, void *oldp, size_t *oldlenp, void *newp, size_t newlen) {
+	int ret;
+
+	if (!(config_prof && opt_prof)) {
+		ret = ENOENT;
+		goto label_return;
+	}
+
+	if (newp != NULL && newlen == sizeof(ssize_t)) {
+		ssize_t max = *(ssize_t *)newp;
+		if (max < -1) {
+			ret = EINVAL;
+			goto label_return;
+		}
+		ssize_t old_max = prof_recent_alloc_max_set(tsd, max);
+		READ(old_max, ssize_t);
+	} else {
+		ssize_t old_max = prof_recent_alloc_max_get(tsd_tsdn(tsd));
+		READ(old_max, ssize_t);
```

Comment:
not: move the READ out of the branches. Also when the write conditions aren’t correct, we should return error code.